app/models/mixins/custom_attribute_mixin.rb

```diff
@@ -21,6 +23,28 @@ module CustomAttributeMixin
         miq_custom_set(custom_str, value)
       end
     end
+
+    def self.custom_keys
+      CustomAttribute.where(:resource_type => base_class).map(&:name).uniq
+    end
+
+    def self.load_custom_attributes_for(cols)
+      custom_attributes = CustomAttributeMixin.select_virtual_custom_attributes(cols)
+      custom_attributes.each { |custom_attribute| add_custom_attribute(custom_attribute) }
+    end
+
+    def self.add_custom_attribute(custom_attribute)
+      virtual_column(custom_attribute.to_sym, :type => :string, :uses => :custom_attributes)
+
+      define_method(custom_attribute.to_sym) do
+        custom_attribute_without_prefix = custom_attribute.sub(CUSTOM_ATTRIBUTES_PREFIX, "")
+        custom_attributes.select { |x| custom_attribute_without_prefix.casecmp(x.name.downcase) == 0 }.shift.try(:value)
```

Comment:
ASIDE: With `pluck()`, you typically don't need to also use `select()`.

This query is known. I was thinking you could use the `select {}` into sql so you'd still have 1 query, but only the rows you want are brought back.

>basically it is about counts of queries vs memory

I may be misunderstanding.
I am hoping moving the select into sql will reduce the number of un-needeed records coming back and making a single query that brings back all the same records.

Is the `value` very complex to determine, or can we do something simple like `where("lower(value) = ?", x.name.downcase)`?